internal/checker/checker.go

```diff
@@ -41,89 +44,128 @@ func New(url, payload string, errors []string, client *resty.Client) Checker {
+	forms, countBefore, err := c.fetchForms(link)
+	fmt.Println("Forms received...")
+	if err != nil {
+		return err
+	}
+	var wg sync.WaitGroup
 
-		for _, payload := range payloads {
-			wg.Add(1)
-			go c.formSubmit(&wg, payload, countBefore)
-		}
-		wg.Wait()
+	scan := bufio.NewScanner(f)
+	for scan.Scan() {
+		payload := scan.Text()
+		wg.Add(1)
+		go c.submitForm(link, payload, countBefore, forms, setValues, &wg)
 	}
+	wg.Wait()
+	fmt.Println("Finished!")
+	return nil
 }
 
 // countErrs counts inclusions of errors from checker error-list if they are existed.
 func (c *checker) countErrs(bytes []byte) int {
-	var count = 0
+	var counter = 0
 	body := string(bytes)
 	for _, err := range c.errors {
-		count += strings.Count(body, err)
+		counter += strings.Count(body, err)
 	}
 
-	return count
+	return counter
 }
 
-// formSubmit submitting form putting each payload. Matches request body with possible errors.
-func (c *checker) formSubmit(wg *sync.WaitGroup, payload string, countBefore int) {
-	defer wg.Done()
-	resp, err := c.client.R().
-		SetFormData(map[string]string{
-			"login":    payload,
-			"password": "12345",
-			"form":     "submit",
-		}).
-		Post(c.url)
+//fetchForms fetches all forms which are exists in given link.
+func (c *checker) fetchForms(link string) ([]*htmlForm.HtmlForm, int, error) {
+	resp, err := c.client.R().Get(link)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error fetching url %q: %v", link, err)
+	}
+
+	root, err := html.Parse(bytes.NewReader(resp.Body()))
 	if err != nil {
-		log.Printf("Error sending form data : %v\n", err)
+		return nil, 0, fmt.Errorf("error parsing response: %v", err)
 	}
-	countAfter := c.countErrs(resp.Body())
 
-	for i, re := range c.errRegexes {
-		if re.MatchString(string(resp.Body())) && countBefore != countAfter {
-			fmt.Printf("FOUND VULNARABILITY IN [%s] TO PAYLOAD [%s] IN URL [%s]\n", c.errors[i], payload, c.url)
-			break
+	forms := htmlForm.ParseForms(root)
+
+	if len(forms) == 0 {
+		return nil, 0, fmt.Errorf("no forms found at %q", link)
+	}
+
+	for _, f := range forms {
+		err := f.ParseURL(link)
+		if err != nil {
+			return forms, 0, err
+		}
+	}
+
+	countBefore := c.countErrs(resp.Body())
+
+	return forms, countBefore, nil
+}
+
+// submitForm submitting htmlForm putting each payload. Matches request body with possible errors.
+func (c *checker) submitForm(link, payload string, countBefore int, forms []*htmlForm.HtmlForm, setValues func(url.Values, string), wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	for _, f := range forms {
+		// filling the form empty attributes with payloads
+		if setValues != nil {
+			setValues(f.Values, payload)
+		}
+
+		resp, err := c.client.R().
+			SetFormDataFromValues(f.Values).
+			Post(f.GetURL())
+		if err != nil {
+			log.Printf("error posting htmlForm: %v", err)
+		}
+
+		countAfter := c.countErrs(resp.Body())
+
+		for i, re := range c.errRegexes {
+			if re.MatchString(string(resp.Body())) && countBefore != countAfter {
```

Comment:
You can move ` string(resp.Body()) ` above for performance improve to some variable.